ecs-agent/netlib/platform/firecracker_linux.go

```diff
@@ -65,7 +68,21 @@ func (f *firecraker) ConfigureInterface(
 	iface *networkinterface.NetworkInterface,
 	netDAO netlibdata.NetworkDataClient,
 ) error {
-	return f.common.configureInterface(ctx, netNSPath, iface, netDAO)
+	var err error
+	switch iface.InterfaceAssociationProtocol {
```

Comment:
why did we switch from using `configureInterface()` method? What's different here?

Reply:
Previously the common method was configuring a TAP interface after the VLAN IF. Now, we have a firecracker-specific method that does the TAP setup, while the common one does the bridge. The logic being that all linux platforms that are non-firecracker will configure branch ENIs through that common method.

Reply:
So, is `configureInterface()` useful anymore? Also, please add that comment in the code.

Reply:
I think `configureInterface()` in the common package is useful since it allows all the non-Firecracker linux platforms to re-use it.